### PR TITLE
Remove kernel builder pattern for Python: sk.kernel_create() -> sk.Kernel()

### PR DESCRIPTION
**Pull request by awharrison-28**
### Motivation and Context

This PR makes creating a kernel instance more pythonic. The builder pattern is removed and the kernel is constructed using Kernel(...).

Work originally started with #137

### Description

- Removed kernel_builder.py.
- Instance creation pattern `kernel = sk.create_kernel() -> kernel = sk.Kernel()`
- For notebook 6, there is no need to create a separate kernel for the GitHub files. Changed to use a separate collection. 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: python/tests/unit/core_skills/test_time_skill.py
Comment (dluc) [inline python/tests/unit/core_skills/test_time_skill.py:1]: would it be possible to say "import everything" and change `kernel = sk.Kernel()` to `kernel = Kernel()`? thoughts?

**Review by alexchaomander — APPROVED** [review]
LGTM!

File: samples/notebooks/python/06-memory-and-embeddings.ipynb [hunk 1]
```diff
@@ -330,29 +321,12 @@
...
-   ]
-  },
-  {
+   "attachments": {},
    "cell_type": "markdown",
    "id": "75f3ea5e",
    "metadata": {},
    "source": [
-    "Now let's add these files to our VolatileMemoryStore using `SaveReferenceAsync`."
+    "Now let's add these files to our VolatileMemoryStore using SaveReferenceAsync. We'll separate these memories from the chat memories by putting them in a different memory collection."
```
Comment (dluc) [inline samples/notebooks/python/06-memory-and-embeddings.ipynb:329]: nit: the word "memory" is repeated several time. maybe "in a different memory collection" => in a different collection."

also pls word wrap at column ~80/100

**Review by dluc — CHANGES_REQUESTED** [review]
pls bump the pip package version number too

File: python/tests/unit/core_skills/test_time_skill.py
Comment (alexchaomander) [reply python/tests/unit/core_skills/test_time_skill.py:1]: It's an anti-pattern for Python to "import everything."  You usually just import what you actually use

**Review by mkarle — CHANGES_REQUESTED** [review]
There's still a kernel_builder usage in python/tests/end-to-end/memory.py.  We also should remove python/semantic_kernel/kernel_builder.py, right?

File: python/tests/unit/core_skills/test_time_skill.py
Comment (dluc) [reply python/tests/unit/core_skills/test_time_skill.py:1]: what I mean, is why not just import the Kernel class, and to `sk = Kernel()`?

Comment (awharrison-28) [reply python/tests/unit/core_skills/test_time_skill.py:1]: using kernel = Kernel() where the semantic_kernel import isn't utilized in any other fashion

**Review by dluc — APPROVED** [review]

**Comment by dluc** [comment]
@mkarle I'll merge after your ✅

**Review by alexchaomander — APPROVED** [review]

**Review by mkarle — APPROVED** [review]

**Pull request merged**